### PR TITLE
refactor(cbr): deprecate policy_id and support policy structure

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Deprecate policy_id and support policy structure.
- deprecate 'policy_id' because this parameter only support backup type policy binding.
- support parameter 'policy' to support all type policies binding.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. deprecate parameter 'policy_id'.
2. support parameter 'policy'.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccVault_bindPolicies'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccVault_bindPolicies -timeout 360m -parallel 4
=== RUN   TestAccVault_bindPolicies
=== PAUSE TestAccVault_bindPolicies
=== CONT  TestAccVault_bindPolicies
--- PASS: TestAccVault_bindPolicies (33.64s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       33.738s
```

### CODE REVIEW

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go [hunk 1]
```diff
@@ -391,10 +411,10 @@ func resourceVaultCreate(ctx context.Context, d *schema.ResourceData, meta inter
 		d.SetId(vault.ID)
 	}
 
-	if policy, ok := d.GetOk("policy_id"); ok {
-		_, err := vaults.BindPolicy(client, d.Id(), vaults.BindPolicyOpts{PolicyID: policy.(string)}).Extract()
+	if d.Get("policy").(*schema.Set).Len() > 0 {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: why not use d.GetOk

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go [hunk 2]
```diff
@@ -459,24 +479,21 @@ func setResources(d *schema.ResourceData, vType string, resources []vaults.Resou
...
 	allPages, err := policies.List(client, listOpts).AllPages()
 	if err != nil {
-		return "", fmt.Errorf("error getting policy by vault ID (%s): %s", vaultId, err)
+		return nil, fmt.Errorf("error getting policy by vault ID (%s): %s", vaultId, err)
 	}
 
 	policyList, err := policies.ExtractPolicies(allPages)
 	if err != nil {
-		return "", fmt.Errorf("error extracting vault list: %s", err)
+		return nil, fmt.Errorf("error exting policy list: %s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: should be "extracting"?

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go [hunk 3]
```diff
@@ -485,12 +502,32 @@ func getNumberInGB(megaBytes float64) float64 {
...
 }
 
-func setPolicyId(d *schema.ResourceData, client *golangsdk.ServiceClient) error {
-	policyId, err := getPolicyByVaultId(client, d.Id())
+func flattenPolicies(d *schema.ResourceData, client *golangsdk.ServiceClient) []map[string]interface{} {
+	vaultId := d.Id()
+	policyList, err := getPoliciesByVaultId(client, vaultId)
 	if err != nil {
-		return err
+		log.Printf("[ERROR] error querying CBR policies by vault ID (%s): %v", d.Id(), err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]:
```suggestion
		log.Printf("[ERROR] error querying CBR policies by vault ID (%s): %v", vaultId, err)
```

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccVault_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccVault_ -timeout 360m -parallel 4
=== RUN   TestAccVault_BasicServer
=== PAUSE TestAccVault_BasicServer
=== RUN   TestAccVault_ReplicaServer
=== PAUSE TestAccVault_ReplicaServer
=== RUN   TestAccVault_prePaidServer
=== PAUSE TestAccVault_prePaidServer
=== RUN   TestAccVault_BasicVolume
=== PAUSE TestAccVault_BasicVolume
=== RUN   TestAccVault_BasicTurbo
=== PAUSE TestAccVault_BasicTurbo
=== RUN   TestAccVault_ReplicaTurbo
=== PAUSE TestAccVault_ReplicaTurbo
=== RUN   TestAccVault_AutoBind
=== PAUSE TestAccVault_AutoBind
=== RUN   TestAccVault_bindPolicies
=== PAUSE TestAccVault_bindPolicies
=== CONT  TestAccVault_BasicServer
=== CONT  TestAccVault_BasicTurbo
=== CONT  TestAccVault_bindPolicies
=== CONT  TestAccVault_prePaidServer
--- PASS: TestAccVault_bindPolicies (51.99s)
=== CONT  TestAccVault_AutoBind
--- PASS: TestAccVault_AutoBind (29.72s)
=== CONT  TestAccVault_ReplicaTurbo
--- PASS: TestAccVault_ReplicaTurbo (20.57s)
=== CONT  TestAccVault_ReplicaServer
--- PASS: TestAccVault_ReplicaServer (20.01s)
=== CONT  TestAccVault_BasicVolume
--- PASS: TestAccVault_BasicVolume (336.23s)
--- PASS: TestAccVault_prePaidServer (478.60s)
--- PASS: TestAccVault_BasicServer (508.40s)
--- PASS: TestAccVault_BasicTurbo (891.88s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       891.969s
```

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: done.

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: yeah

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cbr' TESTARGS='-run=TestAccVault_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cbr -v -run=TestAccVault_ -timeout 360m -parallel 4
=== RUN   TestAccVault_BasicServer
=== PAUSE TestAccVault_BasicServer
=== RUN   TestAccVault_ReplicaServer
=== PAUSE TestAccVault_ReplicaServer
=== RUN   TestAccVault_prePaidServer
=== PAUSE TestAccVault_prePaidServer
=== RUN   TestAccVault_BasicVolume
=== PAUSE TestAccVault_BasicVolume
=== RUN   TestAccVault_BasicTurbo
=== PAUSE TestAccVault_BasicTurbo
=== RUN   TestAccVault_ReplicaTurbo
=== PAUSE TestAccVault_ReplicaTurbo
=== RUN   TestAccVault_AutoBind
=== PAUSE TestAccVault_AutoBind
=== RUN   TestAccVault_bindPolicies
=== PAUSE TestAccVault_bindPolicies
=== CONT  TestAccVault_BasicServer
=== CONT  TestAccVault_BasicVolume
=== CONT  TestAccVault_ReplicaTurbo
=== CONT  TestAccVault_bindPolicies
--- PASS: TestAccVault_ReplicaTurbo (25.01s)
=== CONT  TestAccVault_AutoBind
--- PASS: TestAccVault_AutoBind (31.17s)
=== CONT  TestAccVault_prePaidServer
--- PASS: TestAccVault_bindPolicies (58.86s)
=== CONT  TestAccVault_BasicTurbo
--- PASS: TestAccVault_BasicVolume (364.55s)
=== CONT  TestAccVault_ReplicaServer
--- PASS: TestAccVault_ReplicaServer (20.30s)
--- PASS: TestAccVault_BasicServer (491.27s)
--- PASS: TestAccVault_prePaidServer (458.05s)
--- PASS: TestAccVault_BasicTurbo (905.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cbr       964.170s
```

**Review by Jason-Zhang9309 — COMMENTED** [review]
/lgtm

File: huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go [hunk 4]
```diff
@@ -180,6 +193,13 @@ func ResourceVault() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			// Deprecated arguments
+			"policy_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "schema:Deprecated",
```
Comment (ShiChangkuo) [inline huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: this line can be deleted

Comment (Lance52259) [reply huaweicloud/services/cbr/resource_huaweicloud_cbr_vault.go]: done

**Pull request merged**